### DISCUSSION TITLE
Changes to pdf format in "Export Meeting List"?

### DISCUSSION THREAD

**Discussion by AASTL** [code4recovery/pdf#73] [Ideas]
Hello! This is the A.A. Central Office in St. Louis, Missouri www.aastl.org, using the 12 Step Meeting List plugin in a WordPress website.

I am trying to generate a pdf meeting guide, to produce a paper meeting schedule, using the "Export Meeting List" function in the 12 Step Meeting List plugin. 

I had changed some of my Regions from municipalities to counties, trying to make searches easier, with fewer options on the Place list. But, when I generated my meeting guide pdf, I get my new county-based regions, but the addresses for the meetings do not include the towns/municipalities. This definitely won't work for people trying to find those meetings.

Is there a way to change the pdf generator, so that the addresses for the meetings include the city/municipality, not just the street address?

Thank you!

**Comment by gkovats** [comment]
~@AASTL could you share the URL to your feed for recreating the issue?~
Disregard, I've been informed this is publicly available. :/ OK, I'm not a crack specialist, but will look now.

**Reply by gkovats** [reply]
Just confirming your issue, PDF shows:
![image](https://github.com/user-attachments/assets/97ffc6da-816d-43cb-9656-d0747c38a6d2)
Or, `5275 S Lindbergh Blvd`
Where as the feed has a full address of `5275 S Lindbergh Blvd, Sappington, MO 63126, USA` 
Would adding City and State to the address fix your issue?

**Reply by AASTL** [reply]
George,

(This reply is a little bit updated from the reply I sent via email)

Yes, I believe that would make it useable! Everything up to and including the zip code?

If I have a “Region” for every municipality, town, or neighborhood in the area I cover, it’s over 100 regions, and that makes it a tough search when they use that Region drop-down menu (the one labeled "Anywhere" to start).

The issue is exacerbated by the fact that in many small towns that formerly were outside of our area, they now want to be listed with us, in order to show up in the Meeting Guide app! That adds at least a couple of dozen towns or "regions," if I'm giving each town its own region.

Thank you!

**Comment by anchovie91471** [comment]
I've transferred this discussion to our PDF service repository as it relates to it.

**Reply by sflwa** [reply]
@anchovie91471 I'm helping @DFWDesignGuy with NECS and have a similar issue - The office is currently using the CSV export and a local Access database to generate a PDF - See https://github.com/code4recovery/12-step-meeting-list/discussions/1486 for an attachment of what it currently looks like - Similar to what I believe @AASTL  is looking for - while we'd love the CSS coloring like the example pdf, having the ability have the full address AND notes would mean we could use the generator vs doing this manually

**Comment by joshreisner** [comment]
hi @AASTL one quick workaround you could do on your end would be to employ hierarchical regions, eg

* Putnam County
  * Unionville
* St Louis County
  * Affton
  * Concord Village
  * Etc

this will have the effect of changing the region grouping in the PDF for `Concord Village` to `St Lous County: Concord Village`, also this will resolve the inconsistency of some meetings having a county region and others having a city/town

you can do this by going to Meetings > Regions and edit the "parent" value - you should not have to update too many actual meetings

**Reply by DFWDesignGuy** [reply]
Thanks Josh!

**Reply by AASTL** [reply]
Hi Josh!

Thanks so much for looking into my issue!

So, I tried embedding the town as its own region with the county as the parent region. It works very well for the pdf generator/paper meeting guide, but it still leaves un-fixed one of my main problems. That is, I still have about 100 things that show up in my Region drop-down, making it difficult to use. Each of those sub-regions shows up in the drop-down.
So, yeah, I think the change of keeping the town and zip in the address, in the pdf, would still be really helpful?

**Reply by joshreisner** [reply]
ok, maybe this can be an 'option' you can check when you are generating the list?

<img width="334" alt="Screenshot 2024-08-12 at 11 17 49 AM" src="https://github.com/user-attachments/assets/d024a067-bfa1-4846-a6c0-c24ff06f40d1">

the listing above would go from: `Concord Trinity UMC, 5275 S Lindbergh Blvd` to `Concord Trinity UMC, 5275 S Lindbergh Blvd, Sappington, MO 63126, USA`

**Reply by AASTL** [reply]
Josh, that would be super-great, for our needs! And, as an "option" seems it would hurt anyone else?
Thank you!

**Reply by AASTL** [reply]
Also, I don't think we need the "USA", if that saves memory or coding or something - but, if getting rid of it is another kind of problem, that doesn't hurt us, to have it!

**Reply by AASTL** [reply]
Josh, if you don't mind my asking about something different - just cause you're already looking at the pdf generator?
It would be very helpful if each page listed what day of the week you're looking at, rather than just having that when a new day begins, in the list.

I understand that sometimes what sounds like a "fix" can create new issues, or what looks simple to the user can be complicated for the builder, so it that's not do-able at this time, I totally understand!

**Reply by joshreisner** [reply]
yes, having a persistent page header would be wonderful. last time i checked DOMPDF (the PDF library we use) didn't support it, but perhaps @anchovie91471 when you get a chance you might confirm? this would ideally show either day or region, depending on which was selected as the top grouping

**Comment by joshreisner** [comment]
@AASTL we have a checkbox now on pdf.code4recovery.org, please let us know if this works for you
[aastl.pdf](https://github.com/user-attachments/files/16592340/aastl.pdf)

**Comment by AASTL** [comment]
Josh,



The pdf download is now GREAT for our purposes, with the addition of the
long address! I’m now able to tweak the Regions and the addresses to limit
the size of my list of towns and counties, so that an Anywhere vs Region
search is now more useful.



My list of regions is still too long, but maybe I need to reconsider the
way they are defined? It goes back to that issue, that groups three hours
away want to be listed on our website so that they will show up in the
Meeting Guide app.



Anyhoo, thanks a TON for your help with this!!



-Tom



Tom Noerper
Office manager

St. Louis Central Services

314.647.3677

www.aastl.org







*From:* Josh Reisner ***@***.***
*Sent:* Monday, August 12, 2024 5:06 PM
*To:* code4recovery/pdf
*Cc:* AASTL; Mention
*Subject:* Re: [code4recovery/pdf] Changes to pdf format in "Export Meeting
List"? (Discussion #73)



@AASTL <https://github.com/AASTL> we have a checkbox now on
pdf.code4recovery.org, please let us know if this works for you
aastl.pdf <https://github.com/user-attachments/files/16592340/aastl.pdf>

—
Reply to this email directly, view it on GitHub
<https://github.com/code4recovery/pdf/discussions/73#discussioncomment-10319211>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/BJLYSZIK7ABBZVKOCOK5EADZREWU7AVCNFSM6AAAAABMFT3N3CVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZRHEZDCMI>
.
You are receiving this because you were mentioned.Message ID:
***@***.***>

**Reply by anchovie91471** [reply]
@AASTL We have recently added a regions filter to PDF. Now you can choose which regions you want on the PDF.

Would this help solve your issue with the groups that are 3 hours away?

**Comment by AASTL** [comment]
Anthony,



This is great!



Tom N.





Tom Noerper
Office manager

St. Louis Central Services

314.647.3677

www.aastl.org







*From:* Anthony Baggett ***@***.***
*Sent:* Sunday, February 15, 2026 9:37 AM
*To:* code4recovery/pdf
*Cc:* AASTL; Mention
*Subject:* Re: [code4recovery/pdf] Changes to pdf format in "Export Meeting
List"? (Discussion #73)



@AASTL <https://github.com/AASTL> We have recently added a regions filter
to PDF. Now you can choose which regions you want on the PDF.

Would this help solve your issue with the groups that are 3 hours away?

—
Reply to this email directly, view it on GitHub
<https://github.com/code4recovery/pdf/discussions/73#discussioncomment-15812509>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/BJLYSZJRG7PNLX3JR4SDLAT4MCHCVAVCNFSM6AAAAACVGK2GGGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTKOBRGI2TAOI>
.
You are receiving this because you were mentioned.[image: Image removed by
sender.]Message ID: <code4recovery/pdf/repo-discussions/73/comments/
***@***.***>
